gimbal-adapter/src/main/java/com/urbanairship/gimbal/AirshipReadyReceiver.java

```diff
@@ -7,14 +7,34 @@
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.support.annotation.NonNull;
+
+import com.urbanairship.UAirship;
+import com.urbanairship.push.RegistrationListener;
 
 /**
  * Broadcast receiver for Airship Ready events.
  */
 public class AirshipReadyReceiver extends BroadcastReceiver {
 
     @Override
-    public void onReceive(Context context, Intent intent) {
+    public void onReceive(final Context context, Intent intent) {
         GimbalAdapter.shared(context).restore();
+
+        UAirship.shared().getPushManager().addRegistrationListener(new RegistrationListener() {
+            @Override
+            public void onChannelCreated(@NonNull String channelId) {
+                GimbalAdapter.shared(context).updateDeviceAttributes();
+            }
+
+            @Override
+            public void onChannelUpdated(@NonNull String channelId) {
```

Comment:
Maybe call `GimbalAdapter.shared(context).updateDeviceAttributes()` here also?

Reply:
We only need to do it on channel create